src/socketio.quiver.js

```diff
@@ -970,23 +1140,107 @@ QuiverSocialProvider.prototype.changeRoster = function(userId, clientSuffix, inv
 };
 
 /**
- * Interpret messages from the server to this as owner.
- * There are 3 types of messages
- * - Directed messages from endpoints (message)
- * - State information from the server on initialization (state)
- * - Roster change events (users go online/offline) (roster)
+ * @param {!QuiverSocialProvider.server_} server The server through which the
+ *     message was delivered.
+ * @param {Object} msg Encrypted message from the server
+ * @private
+ **/
+QuiverSocialProvider.prototype.onEncryptedMessage_ = function(server, msg) {
+  if (!msg) {
+    return;
+  }
+
+  this.pgp_.verifyDecrypt(msg.cipherText, msg.key).then(function(plain) {
```

Comment:
Exactly.  (Actually, some pings are just signed, but most are also encrypted.)  `onEncryptedMessage_` will happily verify signatures on signed-but-not-encrypted messages.

BTW, even the "cleartext" messages are not "clearsigned", so they're OpenPGP-format binary blobs, not human-readable text.